server/schema.ts

```diff
@@ -33,12 +33,15 @@ export interface ITagDetailItem {
 	checked_in: boolean;
 	checked_in_date: Date;
 	checked_in_by: string;
+	checkin_success: boolean;
 }
 
 export interface ITagItem {
+	checkin_success: boolean;
 	checked_in: boolean;
 	checked_in_date?: Date;
 	checked_in_by?: string;
+	last_successful_checkin?: ITagDetailItem | null;
```

Comment:
Doesn't need a `?` suffix if you have a union with a null. Otherwise the type is `ITagDetailItem | undefined | null` which is probably not what you want.